pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -210,6 +218,38 @@ func (p *LaunchTemplateProvider) createLaunchTemplate(ctx context.Context, optio
 	return output.LaunchTemplate, nil
 }
 
+// hydrateCache queries for existing Launch Templates created by Karpenter for the current cluster and adds to the LT cache.
+// Any error during hydration will result in a panic
+func (p *LaunchTemplateProvider) hydrateCache(ctx context.Context) {
+	queryKey := fmt.Sprintf(launchTemplateNameFormat, injection.GetOptions(ctx).ClusterName, "*")
+	p.logger.Debugf("Hydrating the launch template cache with names matching \"%s\"", queryKey)
+	if err := p.ec2api.DescribeLaunchTemplatesPagesWithContext(ctx, &ec2.DescribeLaunchTemplatesInput{
+		Filters: []*ec2.Filter{{Name: aws.String("launch-template-name"), Values: []*string{aws.String(queryKey)}}},
+	}, func(output *ec2.DescribeLaunchTemplatesOutput, _ bool) bool {
+		for _, lt := range output.LaunchTemplates {
+			p.cache.SetDefault(*lt.LaunchTemplateName, lt)
+		}
+		return true
+	}); err != nil {
+		panic(fmt.Sprintf("Unable to hydrate the AWS launch template cache, %s", err.Error()))
+	}
+	p.logger.Debugf("Finished hydrating the launch template cache with %d items", p.cache.ItemCount())
+}
+
+func (p *LaunchTemplateProvider) onCacheEvicted(key string, lt interface{}) {
+	p.Lock()
+	defer p.Unlock()
+	if _, expiration, _ := p.cache.GetWithExpiration(key); expiration.After(time.Now()) {
+		return
+	}
+	launchTemplate := lt.(*ec2.LaunchTemplate)
+	if _, err := p.ec2api.DeleteLaunchTemplate(&ec2.DeleteLaunchTemplateInput{LaunchTemplateId: launchTemplate.LaunchTemplateId}); err != nil {
+		p.logger.Errorf("Unable to delete launch template, %v", err)
```

Comment:
It's not retried (doesn't reconcile) since this is only executed on cache eviction, BUT it does rehydrate on startup, so if something did happen, a restart of Karpenter would clean them up.